src/libstd/sync/mod.rs

```diff
@@ -42,3 +42,18 @@ mod condvar;
 mod mutex;
 mod once;
 mod rwlock;
+
+/// Save power or switch hyperthreads in a busy-wait spin-loop.  This
+/// function is deliberately more primitive than
+/// `std::thread::yield_now` and does not directly yield to the
+/// system's scheduler.  In some cases it might be useful to use a
+/// combination of both functions.
+#[inline]
+#[stable(feature = "rust1", since = "1.1.6")]
+pub fn spin_loop_pause()
+{
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
[`_mm_pause`](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=pause&expand=3864) is part of SSE2, so we should probably use that. But that is exactly as portable as what's provided here I think. (If `_mm_pause` is the answer, then we should probably punt on this until SIMD.)